dashboard/app/views/videos/embed.html.haml

```diff
@@ -1,6 +1,14 @@
+/ This view creates a page that transparently provides fallback-player support to the specified YouTube video.
+/ See VideosController#embed for usage
+!!! 5
+- video = local_assigns[:video]
+- process_fallback_videos = current_user.try(:admin?) && !(Rails.env.production? || Rails.env.test?)
+- Youtube.process(video.key) if process_fallback_videos
```

Comment:
Could you move the business logic like Youtube video processing into the controller?  This seems too heavyweight for a view and is hard to test properly in that context.